spec/openapi.yaml

```diff
@@ -822,6 +821,383 @@ paths:
         400:
           description: Bad Request. Check the json response for more details on what went wrong.
       deprecated: false
+  /v2-preview/reporting/messages/async/detail:
+    post:
+      security:
+        - basic_auth: [ ]
+        - hmac_auth: [ ]
+      tags:
+        - Messaging Reports
+      summary: Post async detail report
+      description: Generates an asynchronous report listing all sent and/or received messages within a specified time period.
+      operationId: PostAsyncDetailReport
+      parameters:
+        - name: start_date
+          in: query
```

Comment:
Hey Sam,
`in: query` has caught my eye, shouldn't request attributes be in the body? If so we might be needing to go and fix the previous endpoints in a separate change. Looking at the live documentation for examples we have [async v1 requests](https://messagemedia.github.io/documentation/#tag/LEGACY-V1-Reports/operation/SubmitReceivedMessagesDetail) that we know are posts with a request body. In there you can see it shows `REQUEST BODY SCHEMA: application/json` rather than `QUERY PARAMETERS` as in the [get v1 report (instant)](https://messagemedia.github.io/documentation/#tag/LEGACY-V1-Reports/operation/GetReceivedMessagesDetail) which is what we have in our published [v2 post detail report](https://messagemedia.github.io/documentation/#tag/Messaging-Reports/operation/PostDetailReport). In query with the style form points to these being url encoded parameters for the request rather than a json body request.
Let me know if you want to discuss this further on Slack but if you can verify the above, I'd suggest we correct the other endpoints in a subsequent change (with a new ticket) rather than adding to this PR too 🙂

Reply:
Hey Liam,

Have pushed changes to migrate the parameters into the request body for the async detail and summary endpoints. I'll ensure to also keep this format for -321, and will create a subtask for refactoring -318 and -319 🚀